doc: boards: align main board image name with board name

kartben opened the pull request:
As per board porting guidelines, the board image should be named after the board.

Compliance check failure is a false positive -- one of the images is too large, but resizing it would be counter productive at this point (old version would still live in Git history).

sylvioalves commented on boards/espressif/esp_wrover_kit/doc/index.rst:
```diff
@@ -43,7 +43,7 @@ Functional Description
 of the ESP-WROVER-KIT board.
 
-.. figure:: img/esp-wrover-kit-v4.1-layout-front.jpg
+.. figure:: img/esp_wrover_kit.jpg
```
@kartben Sounds you only renamed the image file.  Was the image re-worked as well?

marekmatej commented:
LGTM

thaoluonguw commented on boards/renesas/ek_ra4m2/doc/index.rst:
```diff
@@ -61,7 +61,7 @@ available internal to the RA MCU
 - 32 MB (256 Mb) External Quad-SPI Flash
 
-.. figure:: ek-ra4m2-board.webp
+.. figure:: ek_ra4m2.webp
```
@kartben : Sorry, I haven't understood the problem here.
"boards/< vendor >/<board_name>/"
Is it required to change the name of image is same with <board_name>?
Is my understanding correct?

kartben replied:
Yes. It's a convention that's important as it's what allows to know, in case there are multiple images for a board, which one actually depicts the board. See https://github.com/zephyrproject-rtos/zephyr/pull/79160 and how this particular board shows up with a "wrong" image. Thanks!

kartben committed:
doc: boards: align main board image name with board name

As per board porting guidelines, the board image should be named after
the board.
This commit implements the changes for boards found to be non-compliant.

Signed-off-by: Benjamin Cabé <benjamin@zephyrproject.org>

kartben commented:
hadn't noticed docs CI failure until now, sorry. Compliance check will still be failing and will need super-power merge rights.

kartben commented:
@sylvioalves  please re-approve if you have a chance
